openedx/core/djangoapps/content_libraries/management/commands/migrate_lib_to_learning_core.py

```diff
@@ -0,0 +1,234 @@
+"""
+Command to import Blockstore-backed v2 Libraries to Learning Core data models.
+
+This will hopefully be very short-lived code.
+"""
+from datetime import datetime, timezone
+from xml.etree import ElementTree as ET
+import logging
+
+from django.db import transaction
+from django.core.management import BaseCommand, CommandError
+from django.core.exceptions import ObjectDoesNotExist
+
+from opaque_keys.edx.locator import LibraryLocatorV2
+from openedx.core.djangoapps.content_libraries import api as lib_api
+from openedx.core.djangoapps.content_libraries import models as lib_models
+from openedx.core.djangoapps.content_libraries import constants as lib_constants
+from openedx.core.lib.blockstore_api import (
+    get_bundle,
+    get_bundle_file_data,
+    get_bundle_files_dict,
+)
+from openedx_learning.core.publishing import api as publishing_api
+from openedx_learning.core.components import api as components_api
+from openedx_learning.core.contents import api as contents_api
+
+
+log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Create a new LearningPackage and initialize with contents from Library.
+
+    If you run this and specify a Library that already has a LearningPackage
+    (using -f), this command will delete that LearningPackage and create a new
+    one to associate with the Libary. It does not modify the existing one.
+
+    All the work is done in a transaction, so errors partway through the
+    process shouldn't cause state inconsistency in the database. A partly-
+    imported course *can* cause data to end up in Django Storages.
+    """
+
+    def add_arguments(self, parser):
+        """
+        Add arguments to the argument parser.
+        """
+        parser.add_argument(
+            'library-key',
+            type=LibraryLocatorV2.from_string,
+            help=('Content Library Key to import content from.'),
+        )
+        parser.add_argument(
+            '-f',
+            '--force',
+            action='store_true',
+            default=False,
+        )
+
+    def handle(self, *args, **options):
+        """
+        Does the work of parsing content from Blockstore and writing it into
+        openedx-learning core models (publishing, components, contents).
+        """
```

Comment:
It's looking plausible that this will be ready before edx.org is ready to migrate to Blockstore.

If that happens, would we want to do the 2-step migration, or would we want to update this mgmt command to read from modulestore when a V1 lib key is provided?

Reply:
Probably the latter, depending on time.